src/main/java/fr/zcraft/quartzlib/tools/items/CraftingRecipes.java

```diff
@@ -172,21 +178,23 @@ public static ShapedRecipe shaped(ItemStack result, Material... materials) {
      * Ingredients are automatically indexed, using uppercase letters
      * (A = first ingredient, B = second ingredient, etc.)
      *
+     * @param recipeName The name of the new recipe.
      * @param result    The resulting item of the recipe.
      * @param materials The ingredients for the recipe
      * @return The dummy recipe.
      */
-    public static ShapedRecipe shaped(ItemStack result, MaterialData... materials) {
-        ShapedRecipe recipe = new ShapedRecipe(result)
-                .shape(generateRecipeShape(materials.length));
-
+    public static @NotNull ShapedRecipe shaped(String recipeName, ItemStack result, RecipeChoice... materials) {
         if (materials.length > 9) {
-            throw new IllegalArgumentException("Too many materials for a recipe.");
+            throw new IllegalArgumentException("Too many materials for a recipe (a maximum of 9 is expected).");
```

Comment:
Pense a mettre avec un I.t("Too many materials for a recipe (a maximum of 9 is expected).")

Reply:
Pour le coup je pense pas ? Normalement les messages d'exception (i.e. les bugs de code) ne doivent pas être traduits, vu qu'ils n'apparaissent pas à l'utilisateur normalement, seulement au développeur.

Reply:
+1, on ne traduit pas les messages des exceptions